paella-matterhorn/javascript/05_initdelegate.js

```diff
@@ -90,6 +90,27 @@ function initPaellaMatterhorn(episodeId, onSuccess, onError) {
 	);
 }
 
+ // ------------------------------------------------------------
+// #DCE(karen): START transform search/series result into series/series format
+// This tranforms the series data into the expected upstream series format
+var searchSeriesToSeriesSeries  = function (searchSeries) {
+    var dcObject = {};
+    for (var key in searchSeries) {
+        // trim out "dc" and lower case first letter
+        var keyTrimmed = key.replace(/^dc/, '');
+        keyTrimmed = keyTrimmed.charAt(0).toLowerCase() + keyTrimmed.slice(1);
+        dcObject[keyTrimmed] =[ {
+            "value": searchSeries[key]
+        }];
+    }
+    if (!paella.matterhorn.serie) {
+        paella.matterhorn.serie = [];
+    }
+    paella.matterhorn.serie[ 'http://purl.org/dc/terms/'] = dcObject;
```

Comment:
Couldn't it just work like this?

`paella.matterhorn.serie[ 'http://purl.org/dc/terms/'] = searchSeriesToSeriesSeries(data[ 'search-results'].result);`

That would make `searchSeriesToSeriesSeries` very easy to reason about – you'd know it would not be changing any state. It'd just give you back a transformed object.